ThePensionsRegulator.Umbraco.Tests/TokenListTests.cs

```diff
@@ -0,0 +1,196 @@
+﻿using Moq;
+using ThePensionsRegulator.Umbraco.Testing;
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+namespace ThePensionsRegulator.Umbraco.Tests
+{
+	[TestFixture]
+	public class TokenListTests
+	{
+		private const string PROPERTY_ALIAS = "alias";
+
+		private (TokenList TokenList, Mock<IPublishedElement> PublishedElement) CreateReadOnlyTokenList()
+		{
+			var publishedElement = UmbracoContentFactory.CreateContent<IPublishedElement>();
+			publishedElement.SetupUmbracoTextboxPropertyValue(PROPERTY_ALIAS, "  example-a example-b    example-c  ");
+			return (new TokenList(publishedElement.Object, PROPERTY_ALIAS), publishedElement);
+		}
+
+		private (TokenList TokenList, Mock<IOverridablePublishedElement> Settings) CreateOverridableTokenList()
+		{
+			var publishedElement = UmbracoContentFactory.CreateContent<IOverridablePublishedElement>();
+			publishedElement.SetupUmbracoTextboxPropertyValue(PROPERTY_ALIAS, "  example-a example-b    example-c  ");
+			return (new TokenList(publishedElement.Object, PROPERTY_ALIAS), publishedElement);
+		}
+
+		[Test]
+		public void Null_publishedElement_returns_empty_TokenList()
+		{
+			var list = new TokenList(null, PROPERTY_ALIAS);
+
+			Assert.That(list.Count, Is.EqualTo(0));
+			Assert.That(list.ToString(), Is.EqualTo(string.Empty));
+		}
+
+		[Test]
+		public void Class_list_is_tokenised()
+		{
+			var tokenList = CreateOverridableTokenList();
+
+			Assert.That(tokenList.TokenList.Count, Is.EqualTo(3));
+			Assert.That(tokenList.TokenList[0], Is.EqualTo("example-a"));
+			Assert.That(tokenList.TokenList[1], Is.EqualTo("example-b"));
+			Assert.That(tokenList.TokenList[2], Is.EqualTo("example-c"));
+		}
+
+		[TestCase("example-b", true)]
+		[TestCase("example-d", false)]
+		[TestCase("example", false)]
+		public void Contains_checks_whole_tokens_only(string checkFor, bool expected)
+		{
+			var tokenList = CreateOverridableTokenList();
+
+			var result = tokenList.TokenList.Contains(checkFor);
+
+			Assert.That(result, Is.EqualTo(expected));
+		}
+
+		public void Can_copy_to_array()
+		{
+			var tokenList = CreateOverridableTokenList();
+
+			var result = new string[tokenList.TokenList.Count];
+			tokenList.TokenList.CopyTo(result, 0);
+
+			Assert.That(result[0], Is.EqualTo("example-a"));
+			Assert.That(result[1], Is.EqualTo("example-b"));
+			Assert.That(result[2], Is.EqualTo("example-c"));
+		}
+
+		[Test]
+		public void Can_find_by_index()
```

Comment:
This tests that an index is returned for a given class, not that the correct class is found by it's index.

Reply:
Well spotted. Fixed.